src/aws-task-scheduler/aws-task-scheduler.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { AwsProviderService } from 'src/provider/aws/aws-provider.service';
+
+@Injectable()
+export class AwsTaskSchedulerService {
+    private readonly logger = new Logger(AwsTaskSchedulerService.name);
+
+    @Cron('45 * * * * *')
+    handleAwsScheduler() {
+        this.logger.debug("This was called to check the Shutdown Time for the VMs");
+
+        const now = new Date();
+        const hours = now.getHours();
+        const minutes = now.getMinutes();
+    
+        if (hours === 23 && minutes === 59) {
```

Comment:
Dieser Block ist nicht notwendig. Da der Scheduler nach Crontab-Syntax geschrieben wird (45 * * ...) triggert er selbstständig am richtigen Zeitpunkt. Im Shutdown muss nur geprüft werden, welche Instanzen alle laufen und diese müssen danm herunter gefahren werden. Nutze dafür einfach den vorhandenen AwsProviderService.

Und schau mal ob man die Konfiguration des Schedulers auslagern kann, um die Shutdownzeit variabel zu halten (jetzt ist sie ja fest einprogrammiert in der Annotation).